sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,521 @@
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+import numpy as np
+
+from sklearn.base import is_classifier
+from sklearn.utils.multiclass import type_of_target
+
+from ...utils._plotting import (
+    _BinaryClassifierCurveDisplayMixin,
+    _despine,
+    _validate_style_kwargs,
+)
+
+
+class CAPCurveDisplay(_BinaryClassifierCurveDisplayMixin):
+    """Cumulative Accuracy Profile (CAP) Curve visualization.
+
+    It is recommended to use
+    :func:`~sklearn.metrics.CAPCurveDisplay.from_estimator` or
+    :func:`~sklearn.metrics.CAPCurveDisplay.from_predictions` to create
+    a :class:`~sklearn.metrics.CAPCurveDisplay`. All parameters are
+    stored as attributes.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    y_true_cumulative : ndarray
+        Cumulative number of true positives, ordered by predictions. If
+        `sample_weight is not `None`, the positive cases are multiplied by
+        their weight before being included into the cumulative sum.
+
+    cumulative_total : ndarray
+        Cumulative number or fraction of cases examined, ordered by
+        predictions. If `sample_weight` is `None`, each case contributes
+        equally. Otherwise this is the (possibly normalized) cumulative sum
+        of weights.
+
+    estimator_name : str, default=None
+        Name of estimator. If None, the estimator name is not shown.
+
+    pos_label : int, float, bool or str, default=None
+        The class considered as the positive class when computing the metrics.
+        By default, `estimators.classes_[1]` is considered as the positive class.
```

Comment:
I see we make this mistake elsewhere (e.g. `RocCurveDisplay`), but I am not sure we should allow `pos_label=None`.
First, if we don't have access to `estimators` we can't infer this. Also we always pass a pos_label in when instantiating the class in `from_estimator` and `from_predictions`.